Problem: can't get a copy of zconfig

vyskocilm opened the pull request:
Solution: add `zconfig_dup` method, with the same semantics wrt NULL
handling as `zmsg_dup` has.

@bluca I did not mark it as draft, I'd say `dup` methods are pretty standard in CLASS/czmq, so there is nothing surprising on this method.

bluca commented on include/czmq_library.h:
```diff
@@ -54,11 +54,12 @@
+#       define CZMQ_EXPORT __attribute__ ((visibility ("default")))
 #   else
 #       define CZMQ_PRIVATE
+#       define CZMQ_EXPORT
```
It's from zproject - @vyskocilm please take it out, I need to verify that the exported symbols do not change first

bluca commented on api/zconfig.api:
```diff
@@ -27,6 +27,14 @@
         Destroy a config item and all its children
     </destructor>
 
+    <method name = "dup">
```
Make it draft please

bluca commented:
It's still new so let's follow procedure please

vyskocilm commented:
@sappo @bluca while I don't think that `dup` is so alien method, I've rebased the PR according your comments. So `dup` is draft and I've dropped `zproject` generated changed.

sappo commented:
@vyskocilm thanks :) It's not about alien methods it's about tracking changes to the API.
